### PR TITLE
test: fuzz test for apply_update

### PR DESCRIPTION
**Pull request by thorseraq**

### CODE REVIEW

File: libs/jwst-codec/fuzz/fuzz_targets/apply_update.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#![no_main]
+
+use jwst_codec::{
+    gen_nest_type_from_nest_type, gen_nest_type_from_root, CRDTParam, ManipulateSource, OpType,
```
[the remark this answers is not in the document]
Comment (thorseraq) [reply libs/jwst-codec/fuzz/fuzz_targets/apply_update.rs:4]: False positive, `jwst-codec` has those export members under `#[cfg(fuzzing)]`
```rust
#[cfg(fuzzing)]
pub use doc::{
    gen_nest_type_from_nest_type, gen_nest_type_from_root, CRDTParam, ManipulateSource, OpType,
    YrsMapOps, YrsNestType,
};
```

File: libs/jwst-codec/fuzz/Cargo.toml [hunk 2]
```diff
@@ -12,10 +12,8 @@ rand = "0.8"
 libfuzzer-sys = "0.4"
 lib0 = "0.16.5"
 yrs = "0.16.5"
-jwst-codec = {path = ".." }
 
-#[dependencies.jwst-codec]
-#path = ".."
+jwst-codec = {path = "../../jwst-codec" }
```
Comment (darkskygit) [inline libs/jwst-codec/fuzz/Cargo.toml:16]: ?

File: libs/jwst-codec-util/src/doc_operation/yrs_op/array.rs [hunk 3]
```diff
@@ -0,0 +1,187 @@
+use super::*;
+
+pub fn gen_array_ref_ops(
+) -> HashMap<NestDataOpType, Box<dyn Fn(&yrs::Doc, &YrsNestType, CRDTParam)>> {
+    let mut ops: HashMap<NestDataOpType, Box<dyn Fn(&yrs::Doc, &YrsNestType, CRDTParam)>> =
```
Comment (darkskygit) [inline libs/jwst-codec-util/src/doc_operation/yrs_op/array.rs:5]: consider https://github.com/rust-phf/rust-phf ?

Comment (thorseraq) [reply libs/jwst-codec-util/src/doc_operation/yrs_op/array.rs:5]: LGTM

File: libs/jwst-codec-util/src/doc_operation/types.rs [hunk 4]
```diff
@@ -0,0 +1,63 @@
+use yrs::{ArrayRef, MapRef, TextRef, XmlElementRef, XmlFragmentRef, XmlTextRef};
+
+pub const NEST_DATA_INSERT: &'static str = "insert";
```
Comment (darkskygit) [inline libs/jwst-codec-util/src/doc_operation/types.rs:3]: not need 'static lifetime after rust 1.60+

File: libs/jwst-codec/Cargo.toml [hunk 5]
```diff
@@ -16,6 +16,7 @@ ordered-float = "3.6.0"
 rand = "0.8.5"
 serde_json = "1.0.94"
 thiserror = "1.0.40"
+yrs = "0.16.5"
```
Comment (darkskygit) [inline libs/jwst-codec/Cargo.toml:19]: move to dev-dependencies

**Review by darkskygit — APPROVED** [review]

**Pull request merged**